integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/ExcludeMultipleHeadersWithCacheTestCase.java

```diff
@@ -0,0 +1,108 @@
+        assertNotNull(stockResponse, "Response is null");
+        String change = stockResponse.getFirstElement().getFirstChildWithName(new QName
+                ("http://services.samples/xsd", "change")).getText();
+
+        // Sending a request to SimpleStockQuote Service with 'user' header
+        axis2Client.addHttpHeader("user", "Peter");
+        stockResponse = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("excludeMultipleHeaders"), "", "ExcludeMultipleHeaders");
+        assertNotNull(stockResponse, "Response is null");
+        String change1 = stockResponse.getFirstElement().getFirstChildWithName(new QName
+                ("http://services.samples/xsd", "change")).getText();
+
+        assertTrue(change.equals(change1), "Response caching did not exclude the given header value");
+
+        // Sending a request to SimpleStockQuote Service with 'country' header
+        axis2Client.addHttpHeader("COUNTRY", "UK");
+        stockResponse = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("excludeMultipleHeaders"), "", "ExcludeMultipleHeaders");
+        assertNotNull(stockResponse, "Response is null");
+        String change2 = stockResponse.getFirstElement().getFirstChildWithName(new QName
+                ("http://services.samples/xsd", "change")).getText();
+
+        assertTrue(change.equals(change2), "Response caching did not works when the letter case is different");
+
+        // Sending a request to SimpleStockQuote Service with 'user' and 'country' headers
+        axis2Client.addHttpHeader("user", "Peter");
+        axis2Client.addHttpHeader("country", "UK");
+        stockResponse = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("excludeMultipleHeaders"), "", "ExcludeMultipleHeaders");
+        assertNotNull(stockResponse, "Response is null");
+        String change3 = stockResponse.getFirstElement().getFirstChildWithName(new QName
+                ("http://services.samples/xsd", "change")).getText();
+
+        assertTrue(change.equals(change3), "Response caching did not exclude all the headers defined for headersToExcludeInHash");
+
+        // Sending a request to SimpleStockQuote Service with 'user' , 'country' , 'userID' headers
+        axis2Client.addHttpHeader("user", "Peter");
+        axis2Client.addHttpHeader("country", "UK");
+        axis2Client.addHttpHeader("userID", "U001");
+        stockResponse = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("excludeMultipleHeaders"), "", "ExcludeMultipleHeaders");
+        assertNotNull(stockResponse, "Response is null");
+        String change4 = stockResponse.getFirstElement().getFirstChildWithName(new QName
+                ("http://services.samples/xsd", "change")).getText();
+
+        assertTrue(!change.equals(change4), "Response caching works with the undefined exclude header");
```

Comment:
assert false can be used

Reply:
Changed the assertion to assert not equal